test/refunder-test.ts

```diff
@@ -0,0 +1,295 @@
+
+import { defaultAccounts } from "@ethereum-waffle/provider";
+import { BigNumber } from "@ethersproject/bignumber";
+import { Contract } from "@ethersproject/contracts";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import { expect } from "chai";
+import { ethers } from "hardhat";
+
+import { 
+	INSUFFICIENT_BALANCE, 
+	NOT_AN_OWNER,
+	NOT_REFUNDABLE
+} from './constants/error-messages.json';
+
+const ethToWei = (ether: string) => {
+	return ethers.utils.parseEther(ether);
+}
+
+describe("Refunder", function() {
+
+	let owner: SignerWithAddress;
+	let addr1: SignerWithAddress;
+
+  	let greeter: Contract;
+	let refunder: Contract;
+
+	beforeEach(async () => {
+		[owner, addr1] = await ethers.getSigners(); 
+
+	  	const Greeter = await ethers.getContractFactory("Greeter");
+		greeter = await Greeter.deploy("Hello, world!");
+		await greeter.deployed();
+
+		// console.log('>> deployedGreeter', deployedGreeter.address);
+
+		const Refunder = await ethers.getContractFactory("Refunder");
+		refunder = await Refunder.deploy();
+		await refunder.deployed();
+
+		// console.log('>> deployedRefunder', deployedRefunder.address);
+	})
+
+	describe('Sending ETHs', () => {
+		it("Owner should send ethers to Refunder", async function() {
+		
+			const balanceOfRefunderBefore = await ethers.provider.getBalance(refunder.address);
+			const value = ethToWei('0.5');
+			const res = await owner.sendTransaction({
+				value: value,
+				to: refunder.address
+			});
+	
+			await res.wait();
+	
+			const balanceOfRefunderAfter = await ethers.provider.getBalance(refunder.address);
+	
+			expect(balanceOfRefunderBefore.toString()).to.be.not.eq(balanceOfRefunderAfter.toString());
+			expect(value.toString()).to.be.eq(balanceOfRefunderAfter.toString());
+		});
+	
+		it("Not owner should send ethers to Refunder", async function() {
```

Comment:
- I think that a better name for the test is `Should be able to deposit ethers`

- I think that it will be even more precise if we do the following check:
balanceOfRefunderAfter = balanceOfRefunderBefore + 0.5 ETH